# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1223316729 14400
#      Mon Oct 06 14:12:09 2008 -0400
# Node ID bb3bdcdaa063318319ebd3b091932d46b7019835
# Parent  277218396978303b29d49502590d507afd47dc9f
oct-type-conv.h (octave_type_conv_body): avoid shadow warning from GCC

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-10-06  John W. Eaton  <jwe@octave.org>
+
+	* ov-type-conv.h (octave_type_conv_body): Avoid shadow warning
+	from GCC.
+
 2008-10-06  David Bateman  <dbateman@free.fr>
 
 	* graphics.cc (static bool is_handle (const octave_value&)): Delete.
 	(static octave_value is_handle (const octave_value&)): New function.
 
 2008-10-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Cell.cc (Cell::delete_elements): Don't pass second arg to
diff --git a/src/ov-type-conv.h b/src/ov-type-conv.h
--- a/src/ov-type-conv.h
+++ b/src/ov-type-conv.h
@@ -28,38 +28,38 @@ octave_type_conv_body (const octave_valu
   octave_value retval;
 
   if (t_arg == t_result || arg.class_name () == name)
     {
       retval = arg;
     }
   else
     {
-      octave_base_value::type_conv_fcn cf
+      octave_base_value::type_conv_fcn cf1
         = octave_value_typeinfo::lookup_type_conv_op (t_arg, t_result);
        
-      if (cf)
+      if (cf1)
         {
-          octave_base_value *tmp (cf (*(arg.internal_rep ())));
+          octave_base_value *tmp (cf1 (*(arg.internal_rep ())));
 
           if (tmp)
             {
               retval = octave_value (tmp);
 
               retval.maybe_mutate ();
             }
         }
       else
         {
-          octave_base_value::type_conv_fcn cf
+          octave_base_value::type_conv_fcn cf2
             = arg.numeric_conversion_function ();
 
-          if (cf)
+          if (cf2)
             {
-              octave_base_value *tmp (cf (*(arg.internal_rep ())));
+              octave_base_value *tmp (cf2 (*(arg.internal_rep ())));
 
               if (tmp)
                 {
                   octave_value xarg (tmp);
 
                   retval = octave_type_conv_body (xarg, name, t_result);
                 }
             }
